lib/utils/blockchain-txs.ts

```diff
@@ -499,14 +549,24 @@ interface NamePrice {
   premium: bigint;
 }
 
+interface GetNamePriceParams {
+  ensName: ENSName;
+  durationInYears: bigint;
+  publicClient: PublicClient & ClientWithEns;
+}
+
 export const getNamePrice = async ({
   ensName,
   durationInYears,
-}: {
-  ensName: ENSName;
-  durationInYears: bigint;
-}): Promise<bigint> => {
+  publicClient,
+}: GetNamePriceParams): Promise<bigint> => {
   const ensNameDirectSubname = ensName.name.split(".eth")[0];
+
+  const chain = publicClient.chain;
+
+  const nameRegistrationContracts =
+    nameRegistrationSmartContracts[chain.id as SupportedNetwork];
```

Comment:
Please refer to my last comment about SupportedNetwork type aliasing and refactor the instances needed to check if the informed chains are supported